File: 프로그래머스/0/181897. 리스트 자르기/리스트 자르기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, vector<int> slicer, vector<int> num_list) {
    vector<int> answer;
    int start, end, gap = 1;

    if (n == 1) {
        start = 0, end = slicer[1];
    } else if (n == 2) {
        start = slicer[0], end = num_list.size()-1;
    } else if (n == 3) {
        start = slicer[0], end = slicer[1];
    } else {
        start = slicer[0], end = slicer[1], gap = 2;
    }

    for (int i = start; i <= end; i+=gap) {
        answer.push_back(num_list[i]);
    }

    return answer;
}

File: 프로그래머스/0/181897. 리스트 자르기/리스트 자르기.swift
import Foundation

func solution(_ n:Int, _ slicer:[Int], _ num_list:[Int]) -> [Int] {
    var start = 0
    var last = 0
    var diff = 1
    var result: [Int] = []
    
    switch n {
    case 1:
        last = slicer[1]
    case 2:
        start = slicer[0]
        last = num_list.count - 1
    case 3:
        start = slicer[0]
        last = slicer[1]
    default:
        start = slicer[0]
        last = slicer[1]
        diff = slicer[2]
    }
    
    while start <= last {
        result.append(num_list[start])
        start += diff
    }
    
    return result
}
